components/brave_wallet/common/cashaddr_unittest.cc

```diff
@@ -0,0 +1,210 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/common/cashaddr.h"
+
+#include "base/strings/string_number_conversions.h"
+#include "brave/components/brave_wallet/common/test_utils.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+
+namespace brave_wallet::cashaddr {
+
+static std::string to_lowercase(const std::string &str) {
+    std::string ret;
+    for(unsigned char c: str) {
+        ret.push_back(std::tolower(c));
+    }
+
+    return ret;
+}
+
+static int char2int(char input) {
+    if(input >= '0' && input <= '9') {
+        return input - '0';
+    }
+    if(input >= 'A' && input <= 'F') {
+        return input - 'A' + 10;
+    }
+    if(input >= 'a' && input <= 'f') {
+        return input - 'a' + 10;
+    }
+    assert(false);
+    return 0; // unreachable
+}
+
+// This function assumes hexstr to be a  sanitized string with
+// an even number of [0-9a-f] characters
+static std::vector<uint8_t> hex2bin(const std::string &hexstr) {
+    std::vector<uint8_t> ret;
+    size_t count = 0;
+    uint8_t first_hex_digit;
+    for (char const &c: hexstr) {
+        if (count % 2) {
+            ret.push_back(first_hex_digit + char2int(c));
+        } else {
+            first_hex_digit = char2int(c) << 4;
+        }
+        count++;
+    }
+    return ret;
+}
+
+static bool are_vectors_equal(const std::vector<uint8_t> vec1,
+                              const std::vector<uint8_t> vec2) {
+    if(vec1.size() != vec2.size()) {
+        return false;
+    }
+    for(size_t i = 0; i < vec1.size(); i ++ ) {
+        if(vec1[i] != vec2[i]) {
+            return false;
+        }
+    }
+    return true;
+}
+
```

Comment:
This is not needed. 
Instead `EXPECT_TRUE(are_vectors_equal(payload, decoded_payload));`
it should be `EXPECT_TRUE(payload, decoded_payload);`